cajita/unit_test/tstSparseDimPartitioner.hpp

```diff
@@ -0,0 +1,293 @@
+    std::array<int, 3> periodic_dims = { 0, 0, 0 };
+    int reordered_cart_ranks = 1;
+    MPI_Comm cart_comm;
+    MPI_Cart_create( MPI_COMM_WORLD, 3, ranks_per_dim.data(),
+                     periodic_dims.data(), reordered_cart_ranks, &cart_comm );
+    int linear_rank;
+    MPI_Comm_rank( cart_comm, &linear_rank );
+    // make a new communicater with MPI_Cart_create
+    MPI_Cart_coords( cart_comm, linear_rank, 3, cart_rank.data() );
+
+    auto owned_cells_per_dim = partitioner.ownedCellsPerDimension( cart_comm );
+    auto owned_tiles_per_dim = partitioner.ownedTilesPerDimension( cart_comm );
+    for ( int d = 0; d < 3; ++d )
+    {
+        auto gt_tile = rec_partitions[d][cart_rank[d] + 1] -
+                       rec_partitions[d][cart_rank[d]];
+        EXPECT_EQ( owned_tiles_per_dim[d], gt_tile );
+        EXPECT_EQ( owned_cells_per_dim[d], gt_tile * cell_per_tile_dim *
+                                               cell_per_tile_dim *
+                                               cell_per_tile_dim );
+    }
+
+    // initialize sparseMap, register every tile on every MPI rank
+    // basic settings for sparseMap
+    double cell_size = 0.1;
+    int pre_alloc_size = size_per_dim * size_per_dim;
+    std::array<double, 3> global_low_corner = { 1.2, 3.3, -2.8 };
+    std::array<double, 3> global_high_corner = {
+        global_low_corner[0] + cell_size * global_cells_per_dim[0],
+        global_low_corner[1] + cell_size * global_cells_per_dim[1],
+        global_low_corner[2] + cell_size * global_cells_per_dim[2] };
+    auto global_mesh = createSparseGlobalMesh(
+        global_low_corner, global_high_corner, global_cells_per_dim );
+    // create a new sparseMap
+    auto sis = createSparseMap<TEST_EXECSPACE>( global_mesh, pre_alloc_size );
+    // register tiles to the sparseMap
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<TEST_EXECSPACE>( 0, size_per_dim ),
+        KOKKOS_LAMBDA( int i ) {
+            for ( int j = 0; j < size_per_dim; j++ )
+                for ( int k = 0; k < size_per_dim; k++ )
+                {
+                    sis.insertCell( i, j, k );
+                }
+        } );
+
+    // compute workload and do partition optimization
+    partitioner.computeLocalWorkLoad( sis );
+    partitioner.computeFullPrefixSum( MPI_COMM_WORLD );
+    partitioner.optimizePartition();
+
+    // check results (should be the same as the average partition)
+    owned_cells_per_dim = partitioner.ownedCellsPerDimension( cart_comm );
+    for ( int d = 0; d < 3; ++d )
+    {
+        auto gt_tile = rec_partitions[d][cart_rank[d] + 1] -
+                       rec_partitions[d][cart_rank[d]];
+
+        EXPECT_EQ( owned_cells_per_dim[d], gt_tile * cell_per_tile_dim *
+                                               cell_per_tile_dim *
+                                               cell_per_tile_dim );
+    }
+}
+
+void biased_distribution_automatic_rank( std::array<int, 3> occupy_start,
```

Comment:
How non-uniform is this test? I think I'll need to discuss with you to understand